u_parrots_test.go

```diff
@@ -0,0 +1,157 @@
+
+	compareClientHelloFields(t, "Vers", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+	compareClientHelloFields(t, "CipherSuites", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+	compareClientHelloFields(t, "CompressionMethods", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+	compareClientHelloFields(t, "NextProtoNeg", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+	compareClientHelloFields(t, "ServerName", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+	compareClientHelloFields(t, "OcspStapling", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+	compareClientHelloFields(t, "Scts", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+	compareClientHelloFields(t, "SupportedCurves", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+	compareClientHelloFields(t, "SupportedPoints", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+	compareClientHelloFields(t, "TicketSupported", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+	compareClientHelloFields(t, "SupportedSignatureAlgorithms", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+	compareClientHelloFields(t, "SecureRenegotiation", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+	compareClientHelloFields(t, "SecureRenegotiationSupported", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+	compareClientHelloFields(t, "AlpnProtocols", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+	compareClientHelloFields(t, "SupportedSignatureAlgorithmsCert", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+	compareClientHelloFields(t, "SupportedVersions", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+	compareClientHelloFields(t, "KeyShares", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+	compareClientHelloFields(t, "EarlyData", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+	compareClientHelloFields(t, "PskModes", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+	compareClientHelloFields(t, "PskIdentities", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+	compareClientHelloFields(t, "PskBinders", uconn.HandshakeState.Hello, generatedUConn.HandshakeState.Hello)
+
+}
+
+func TestUTLSFingerprintClientHello(t *testing.T) {
+	checkUTLSFingerPrintClientHello(t, HelloChrome_58)
+	checkUTLSFingerPrintClientHello(t, HelloChrome_70)
+	checkUTLSFingerPrintClientHello(t, HelloChrome_83)
+	checkUTLSFingerPrintClientHello(t, HelloFirefox_55)
+	checkUTLSFingerPrintClientHello(t, HelloFirefox_63)
+	checkUTLSFingerPrintClientHello(t, HelloIOS_11_1)
+	checkUTLSFingerPrintClientHello(t, HelloIOS_12_1)
+	checkUTLSFingerPrintClientHello(t, HelloRandomized)
+	checkUTLSFingerPrintClientHello(t, HelloRandomizedALPN)
+	checkUTLSFingerPrintClientHello(t, HelloRandomizedNoALPN)
```

Comment:
Nice these are some really awesome tests (commenting on your whole testing process).

Reply:
Thanks! Yeah I'd like to add some actual live checks against the openssl server (as it currently does with the other client hello specs), but I feel good about this being a minimal set.